### PR TITLE
Add genesis configuration isPoa() convenience function, and use it in validation of `--miner-enabled` option

### PR DESCRIPTION
**Pull request by matthew1001**
## PR description
This PR indirectly relates to issue https://github.com/hyperledger/besu/issues/5528.

The intention of the PR is to change the validation logic of the `--miner-enabled` option, particularly with regard to `--min-gas-price` option which is a common requirement for private POA chains.

Specifically, in the case where a POA consensus algorithm has been configured (`qbft`, `ibft2`, `ibft`, or `clique`) validation of options that would otherwise require `--miner-enabled` to be set, is skipped.

## Fixed Issue(s)
No specific issue is open for this.

### CODE REVIEW

File: config/src/main/java/org/hyperledger/besu/config/GenesisConfigOptions.java [hunk 1]
```diff
@@ -65,6 +65,13 @@ public interface GenesisConfigOptions {
    */
   boolean isClique();
 
+  /**
+   * Is poa boolean.
```
Comment (siladu) [inline config/src/main/java/org/hyperledger/besu/config/GenesisConfigOptions.java:69]:
```suggestion
   * Is a Proof of Authority network.
```

File: besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java [hunk 2]
```diff
@@ -2133,7 +2134,10 @@ private void issueOptionWarnings() {
             "--remote-connections-max-percentage"));
 
     // Check that block producer options work
-    if (!isMergeEnabled()) {
+    if (!isMergeEnabled()
+        && !(Optional.ofNullable(genesisConfigOptions)
+            .orElseGet(() -> new StubGenesisConfigOptions())
+            .isPoa())) {
```
Comment (siladu) [inline besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java:2140]: Wonder if we can get away with `!isMergeEnabled && isEthHash` instead?

Note, currently public networks genesis' are either clique or ethash, but future public testnets (proof of stake at genesis) might have the consensus unspecified. These networks shouldn't care about miner-enabled either I think.

Comment (siladu) [reply besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java:2140]: Had a play and I think that `!isMergeEnabled && isEthHash` is what we want for both checks, i.e. the `miner-stratum-enabled` one underneath this (I think it's a bug it doesn't have `!isMergeEnabled`)

PoS at genesis networks that omit the consensus engine will default to "unknown" in the besu code, so it works for them too.

Maybe eventually it can simply be `isEthHash` but might have to always support a "merged" mainnet config (where it was originally ethash).

Comment (siladu) [reply besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java:2140]: ...so yes, the isPoa is probably redundant for this, though potentially useful for other things at some point?

Comment (matthew1001) [reply besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java:2140]: Thanks @siladu yes for this case I think `isEthHash()` makes sense. Maybe it's worth leaving `isPoa()` as a general utility that could help make future code changes/logic a little clearer. I'll submit a new commit that changes the current logic but leaves `isPoa()` in. Let me know what you think.

Comment (matthew1001) [reply besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java]: Realised that `getActualGenesisConfigOptions()` provides the combined output of a genesis file, network config, + overrides, so no refactoring required.

Comment (matthew1001) [reply besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java]: @siladu I think I'm happy with the current shape of the PR:

- switched to using `isEthash()`
- moved `--miner-stratum` option warning check inside same block as `--miner-enabled`
- left `isPoa()` in as a helper function
- added some new unit tests for QBFT & IBFT2 option checking

Ready for you to take another look at when you have a mo.

File: besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java [hunk 3]
```diff
@@ -2133,7 +2133,7 @@ private void issueOptionWarnings() {
             "--remote-connections-max-percentage"));
 
     // Check that block producer options work
-    if (!isMergeEnabled()) {
+    if (!isMergeEnabled() && getActualGenesisConfigOptions().isEthHash()) {
```
Comment (siladu) [inline besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java:2136]: Nice! Didn't know about this one.

File: besu/src/test/java/org/hyperledger/besu/cli/BesuCommandTest.java [hunk 4]
```diff
@@ -3684,7 +3698,7 @@ public void stratumMiningIsEnabledWhenSpecified() throws Exception {
   @Test
   public void stratumMiningOptionsRequiresServiceToBeEnabled() {
 
-    parseCommand("--miner-stratum-enabled");
+    parseCommand("--network", "dev", "--miner-stratum-enabled");
```
Comment (siladu) [inline besu/src/test/java/org/hyperledger/besu/cli/BesuCommandTest.java:3701]: Why was this needed?

**Review by siladu — APPROVED** [review]
Looks good!

**Comment by siladu** [comment]
@matthew1001 Do you have merge permission or need a maintainer to merge it?

File: besu/src/test/java/org/hyperledger/besu/cli/BesuCommandTest.java (hunk 4 above)
Comment (matthew1001) [reply besu/src/test/java/org/hyperledger/besu/cli/BesuCommandTest.java:3701]: I needed to modify the test in a way that would ensure it was being run with a) `merge` enabled and b) `ethash` enabled, so that the warning message it was testing for would be produced. The default network is `mainnet.json` which doesn't have `ethash` enabled but `dev.json` does so I ended up going with that.

**Comment by matthew1001** [comment]
> @matthew1001 Do you have merge permission or need a maintainer to merge it?

No not yet @siladu, still working towards maintainer status so you or one of the other maintainers will need to merge it please.
